src/main.rs

```diff
@@ -376,6 +376,30 @@ async fn enter_redraw_loop(
 					None => needs_redraw = false,
 					Some(action) => match action {
 						InputAction::Redraw => (),
+						InputAction::Click { col, row } => {
+							if let Some((page_num, device_x, device_y)) = tui.map_click_to_page(col, row, &main_area, font_size) {
+								let (resp_tx, resp_rx) = flume::bounded(1);
+								to_renderer.send(RenderNotif::QueryLinkAt {
+									page: page_num,
+									device_x_px: device_x,
+									device_y_px: device_y,
+									resp: resp_tx
+								})?;
+								// await the renderer reply
+								match resp_rx.recv_async().await {
```

Comment:
I think I'd prefer to have this future be awaited within the `tokio::select` block so that we can keep the system as responsive as possible just in case the renderer thread takes a while to respond